Add FAQs from Synapse 1P Migration Office Hours

nick-shapiro opened the pull request:
Incorporating FAQs from Synapse 1P Migration weekly office hours.

nick-shapiro committed:
Add FAQs from Synapse 1P Migration Office Hours

Incorporating FAQs from Synapse 1P Migration weekly office hours.

kalamut commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -75,14 +75,19 @@ sections:
+          Capabilities and requirements are different between the two types of pools. Differences include database objects, query language capabilities, security, tools, data access, and data format. For detailed comparison of SQL pools and serverless pools, please visit [Pools Comparison](https://docs.microsoft.com/en-us/azure/synapse-analytics/sql/overview-features). For best practices when usi ...
+
+      - question: |
+          What are Delta tables, and why should I use them?
```
Could we add a section about how to use auto compaction in delta tables? It is one of the frequently asked questions

nick-shapiro replied:
Added in commit cb70304

kalamut commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -101,7 +106,17 @@ sections:
+      - question: |
+          Can I call notebooks from ADF pipelines?
+        answer: |
+          There are two options for this use case. One option is to keep pipelines in ADF, and you will have to wrap in a web activity. The other option is to migrate the pipelines to Synapse.
```
Could you please add links to each of these options?
Doc for differences between ADF and synapse pipelines:
https://docs.microsoft.com/en-us/azure/synapse-analytics/data-integration/concepts-data-factory-differences

Doc for web activity in ADF:
https://docs.microsoft.com/en-us/azure/data-factory/control-flow-web-activity

Tool for migration:
https://github.com/Azure-Samples/Synapse/tree/main/Pipelines/ImportADFtoSynapse

nick-shapiro replied:
Added in commit cb70304

kalamut commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -75,14 +75,19 @@ sections:
+          Does Azure Synapse workspace Support CI/CD? 
         answer: |
-          Azure Synapse Link for Azure SQL Database is in preview. For more information, see [What is Azure Synapse Link for SQL?](synapse-link/sql-synapse-link-overview.md).
+          Yes! All Pipeline artifacts, notebooks, SQL scripts, and Spark job definitions will reside in Git. All pool definitions will be stored in Git as Azure Resource Manager (ARM) templates. Dedicated SQL pool objects (schemas, tables, views, etc.) will be managed with database projects with CI/CD support.
```
Link to docs https://docs.microsoft.com/en-us/azure/synapse-analytics/cicd/continuous-integration-delivery

nick-shapiro replied:
Added in commit cb70304

kalamut reviewed (changes requested):
Request to add links and update delta section

nick-shapiro committed:
Update overview-faq.yml

Add information for Auto Compaction and links. Addressing comments from @kalamut review.

Jak-MS commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -75,14 +75,24 @@ sections:
+          Does Azure Synapse workspace Support CI/CD? 
         answer: |
-          Azure Synapse Link for Azure SQL Database is in preview. For more information, see [What is Azure Synapse Link for SQL?](synapse-link/sql-synapse-link-overview.md).
+          Yes! All Pipeline artifacts, notebooks, SQL scripts, and Spark job definitions will reside in Git. All pool definitions will be stored in Git as Azure Resource Manager (ARM) templates. Dedicated SQL pool objects (schemas, tables, views, etc.) will be managed with database projects with CI/CD support. For more information, check out this [CI and CD Guide](https://docs.microsoft.com/en-us ...
```
```suggestion
          Yes! All Pipeline artifacts, notebooks, SQL scripts, and Spark job definitions will reside in Git. All pool definitions will be stored in Git as Azure Resource Manager (ARM) templates. Dedicated SQL pool objects (schemas, tables, views, etcetera.) will be managed with database projects with CI/CD support. For more information, check out this [CI and CD Guide](/azure/synapse-analytics/cicd/continuous-integration-delivery).
```

Jak-MS commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -75,14 +75,24 @@ sections:
-          Does Azure Synapse workspace Support CI/CD? 
+          What are the functional differences between dedicated SQL pools and serverless pools?
+        answer: |
+          Capabilities and requirements are different between the two types of pools. Differences include database objects, query language capabilities, security, tools, data access, and data format. For detailed comparison of SQL pools and serverless pools, please visit [Pools Comparison](https://docs.microsoft.com/en-us/azure/synapse-analytics/sql/overview-features). For best practices when usi ...
```
```suggestion
          Capabilities and requirements are different between the two types of pools. Differences include database objects, query language capabilities, security, tools, data access, and data format. For detailed comparison of SQL pools and serverless pools, please visit [Pools Comparison](/azure/synapse-analytics/sql/overview-features). For best practices when using either type of pool, please view [Best Practices for Dedicated SQL Pool](/azure/synapse-analytics/sql/best-practices-dedicated-sql-pool) and [Best Practices for Serverless SQL Pool](/azure/synapse-analytics/sql/best-practices-serverless-sql-pool).
```

Jak-MS commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -95,13 +105,28 @@ sections:
+      - question: |
+          How are Azure Data Factory pipelines and Azure Synapse pipelines different?
+        answer: |
+          Some examples of differences are support for global parameters, monitoring of Spark Jobs for Data Flow, and Integration Runtime sharing. For more information, check out this doc for [Data Integration - Synapse vs ADF](https://docs.microsoft.com/en-us/azure/synapse-analytics/data-integration/concepts-data-factory-differences). 
```
```suggestion
          Some examples of differences are support for global parameters, monitoring of Spark Jobs for Data Flow, and Integration Runtime sharing. For more information, check out this doc for [Data Integration - Synapse vs ADF](/azure/synapse-analytics/data-integration/concepts-data-factory-differences). 
```

Jak-MS commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -95,13 +105,28 @@ sections:
+      - question: |
+          How do I use an Apache Spark Job Definition?
+        answer: |
+          Check out this [Quick Start Guide](https://docs.microsoft.com/en-us/azure/synapse-analytics/quickstart-transform-data-using-spark-job-definition).
```
```suggestion
          Check out this [Quick Start Guide](/azure/synapse-analytics/quickstart-transform-data-using-spark-job-definition).
```

Jak-MS commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -95,13 +105,28 @@ sections:
+      - question: |
+          Can I call notebooks from ADF pipelines?
+        answer: |
+          There are two options for this use case. One option is to keep pipelines in ADF, and you will have to wrap in a web activity. For more information on this option, please view this [Web Activity Guide](h.ttps://docs.microsoft.com/en-us/azure/data-factory/control-flow-web-activity). The other option is to migrate the pipelines to Synapse. For more information on the second option, check o ...
```
```suggestion
          There are two options for this use case. One option is to keep pipelines in ADF, and you will have to wrap in a web activity. For more information on this option, please view this [Web Activity Guide](/azure/data-factory/control-flow-web-activity). The other option is to migrate the pipelines to Synapse. For more information on the second option, check out this [Migration Code Sample](https://github.com/Azure-Samples/Synapse/tree/main/Pipelines/ImportADFtoSynapse).
```

Jak-MS commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -148,7 +173,12 @@ sections:
+      - question: |
+          What tools are available to me on Synapse Spark?
+        answer: |
+           MSSparkUtils on Synapse Spark offers a variety of utilities to enhance your experience and to make integration with other tools and services easier. Work with file systems, get environment variables, chain notebooks together, and work with secrets with minimal manual steps. For full documentation, please visit [Microsoft Spark Utilities](https://docs.microsoft.com/en-us/azure/synapse-a ...
```
```suggestion
           MSSparkUtils on Synapse Spark offers a variety of utilities to enhance your experience and to make integration with other tools and services easier. Work with file systems, get environment variables, chain notebooks together, and work with secrets with minimal manual steps. For full documentation, please visit [Microsoft Spark Utilities](/azure/synapse-analytics/spark/microsoft-spark-utilities?pivots=programming-language-csharp).
```

Jak-MS committed:
fixing absolute links

kalamut commented on articles/synapse-analytics/overview-faq.yml:
```diff
@@ -75,14 +75,24 @@ sections:
+          What is Auto Compaction?
         answer: |
-          Yes! All Pipeline artifacts, notebooks, SQL scripts, and Spark job definitions will reside in Git. All pool definitions will be stored in Git as Azure Resource Manager (ARM) templates. Dedicated SQL pool objects (schemas, tables, views, etc.) will be managed with database projects with CI/CD support.
+          Auto Compaction is one of two complementary features of Auto Optimize for Delta tables. After a write to a table succeeds, Auto Compaction can further compact files for partitions that have the most number of small files. Opting in to Auto Compaction is recommended for streaming use cases where adding minutes of latency is acceptable and when you don't have regular OPTIMIZE calls on you ...
```
@nick-shapiro This is databricks documentation, can you please update Synapse's or remove the link if Synapse does not have one?
